File: blablamovebackend/src/main/java/fr/polytech/unice/blablamove/teamc/blablamovebackend/model/influxdb/RouteCanceled.java
package fr.polytech.unice.blablamove.teamc.blablamovebackend.model.influxdb;

import org.influxdb.annotation.Column;
import org.influxdb.annotation.Measurement;

import java.time.Instant;

/**
 * Class RouteCanceled
 *
 * @author Tanguy Invernizzi
 */
@Measurement(name = "route_canceled")
public class RouteCanceled {

    /**
     * The id of this route.
     */
    @Column(name = "route_uuid")
    private String route_uuid;

    /**
     * The time at which the route has been canceled.
     */
    @Column(name = "time")
    private Instant time;

    public Instant getTime() {
        return time;
    }

    public String getRoute_uuid() {
        return route_uuid;
    }

    public void setRoute_uuid(String route_uuid) {
        this.route_uuid = route_uuid;
    }

    public void setTime(Instant time) {
        this.time = time;
    }
}

File: adminMarketingUI/src/app/model/route-canceled.ts
export class RouteCanceled {

  private _route_uuid: string;
  private _time: Date;

  constructor(route_uuid: string, time: Date) {
    this._route_uuid = route_uuid;
    this._time = time;
  }

  get route_uuid(): string {
    return this._route_uuid;
  }

  set route_uuid(value: string) {
    this._route_uuid = value;
  }

  get time(): Date {
    return this._time;
  }

  set time(value: Date) {
    this._time = value;
  }
}
